test/functional/test_framework/p2p.py

```diff
@@ -217,7 +217,12 @@ def peer_disconnect(self):
     def connection_made(self, transport):
         """asyncio callback when a connection is opened."""
         assert not self._transport
-        logger.debug("Connected & Listening: %s:%d" % (self.dstaddr, self.dstport))
+        info = transport.get_extra_info("socket")
+        us = info.getsockname()
+        them = info.getpeername()
+        logger.debug(f"Connected: us={us[0]}:{us[1]}, them={them[0]}:{them[1]}")
+        self.dstaddr = them[0]
+        self.dstport = them[1]
```

Comment:
0a6797ffb3eb2d45bce1dc68e830efd89ac2a442

Teleporting the comment from https://github.com/bitcoin/bitcoin/pull/29420#discussion_r1592701864

Why didn't we need to do this on master?

The old log message was `Connected & Listening: 127.0.0.1:15689` not `Connected & Listening: 0:0`

Reply:
I replied to that comment to keep the discussion in one place.